### ISSUE TITLE
CSS: Font sizes should be browser responsive

### ISSUE THREAD

**Issue by AnthonySaunders** [joedolson/wp-accessibility#17]
Please change all font-sizes to use "rem" values instead of pixel values, as pixels values are not responsive to default browser settings which are used by those with disabilities for site readability. Particularly the wpa_css() function that uses font-size: 14px for the skip links.

**Comment by joedolson** [comment]
These values are user configurable. 14px offers a reasonable default that I can't provide using rem, due to not having any information about the context the font size is being set in. If you can provide a viable size alternative (in a pull request) that renders satisfactorily in a wide variety of themes, I can implement this. 

Right now, I don't know a way to set the font size in rems without having vastly variable results in different themes. This is largely a problem because the skip links are attached to the `body` element, and it's highly variable whether the root font-size is set on the body element or not.

**Comment by AnthonySaunders** [comment]
When you say the values are user configurable, do you mean from the settings page? 

As far as I can tell, that is not the case and is hard coded into the style block for the skip links.

The settings being user configurable is true for the browser, but pixel size font specifications override default values (i.e. if your default value is set to 16px or 24px or 30px those values are overwritten by the value in the instantiated style block, where as .875rem would make the skip links adjust to the default browser settings). Those with vision impairments of considerable difference to a standard user that may also be using skip links are being neglected in this case, as those skip links are likely unreadable at 14px.

Can you explain why font-size: .875rem an issue? Even if the theme alters the style to make them larger (which would generally be preferred to adjust the buttons to the theme) it doesn't make the skip links stop working or otherwise unusable. These would only be adjusted if the HTML element was specified to a different size other than 16px (even if the body is set to 16px, rem specifically uses the root element of HTML for sizes).

In short, 14px as a font-size is not ADA compliant, so it seems counter intuitive to mandate a specified font-size at the cost of compliance on a plugin that is supposed to help accessibility. This requires developers to go in and update the plugin manually every time it is updated to ensure accessibility is preserved.

Could you perhaps give a setting that allows you to change the font-size for the skip links specifically and simply default it to 14px? That would allow those that wish to use relative sizes (ADA compliant) while keeping the current look and feel for those already using the plugin that do not wish to pursue it.

**Comment by joedolson** [comment]
Yes, in settings. The size is declared in the default focus values, but if you add your own custom focus styles, those would override the defaults. They're appended in the code, but because the custom styles are added later, they'll override the earlier styles. 

The problem with .875 rem is most prominent in sites where the theme is declaring the root font size at 62.5% - a common method to make math simpler by adjusting the rem root to a 10px equivalency. That would make .875 rem an uncomfortably small font size. 

I'm certainly not worried if the font size is adjusted to be larger; but adjusting down to 62.5% is a very common practice.

**Comment by joedolson** [comment]
As a note - if you're needing to adjust the code in WP Accessibility on every install, might I suggest instead adjusting the code in your theme? The whole point of WP Accessibility is to be a low-effort stop gap, not a permanent remedy.

**Comment by AnthonySaunders** [comment]
Multisite installs would require adjusting across every theme that is needing to be adjusted. That's a lot of custom CSS and style block changes within the settings values to accomplish compliance, when adjusting a single value inside of a style block would suffice.

Theoretically, no font-size is even necessary in the case of the skip links, and visual direction is the role of the theme generally (within reason). I understood the primary goal of the plugin was to be accessibility, so visual aesthetics would not be prioritized over that. (the same argument you just made for me to adjust it for myself, is the argument you could make to the ones that want to continue using 14px). 

I just figured as a plugin geared towards accessibility it would make more sense to use the accessible font-size specifications. 

I'll continue to adjust the style block if your stance is to continue using static font-sizes.

**Comment by joedolson** [comment]
I don't have any problem with excluding the font-size; but it will probably result in the font-size getting significantly smaller on a large percentage of sites. However, this is not an area of the plug-in I've specifically tested in years, so I'll certainly take a look at it. 

What I *want* people to do is fix their themes.

**Comment by joedolson** [comment]
Resolved in https://github.com/joedolson/wp-accessibility/commit/b8445fa2fbbfa3085c174b19de32b2a40d9616cc
